src/server/set_family.cc

```diff
@@ -819,70 +832,63 @@ OpResult<StringVec> OpInter(const Transaction* t, EngineShard* es, bool remove_f
   return result;
 }
 
-// count - how many elements to pop.
-OpResult<StringVec> OpPop(const OpArgs& op_args, string_view key, unsigned count) {
-  auto& db_slice = op_args.shard->db_slice();
-  auto find_res = db_slice.FindMutable(op_args.db_cntx, key, OBJ_SET);
+OpResult<StringVec> OpRandMember(const OpArgs& op_args, std::string_view key, int count) {
+  auto find_res = op_args.shard->db_slice().FindReadOnly(op_args.db_cntx, key, OBJ_SET);
   if (!find_res)
     return find_res.status();
 
-  StringVec result;
-  if (count == 0)
-    return result;
+  const CompactObj& co = find_res.value()->second;
 
-  auto it = find_res->it;
-  size_t slen = it->second.Size();
+  const std::size_t size = co.Size();
+  const bool picks_are_unique = count >= 0;
+  const std::size_t picks_count =
+      picks_are_unique ? std::min(static_cast<std::size_t>(count), size) : std::abs(count);
 
-  /* CASE 1:
-   * The number of requested elements is greater than or equal to
-   * the number of elements inside the set: simply return the whole set. */
-  if (count >= slen) {
-    PrimeValue& pv = it->second;
-    if (IsDenseEncoding(pv)) {
-      StringSet* ss = (StringSet*)pv.RObjPtr();
-      ss->set_time(MemberTimeSeconds(op_args.db_cntx.time_now_ms));
-    }
+  std::unique_ptr<PicksGenerator> generator =
+      picks_are_unique ? static_cast<std::unique_ptr<PicksGenerator>>(
+                             std::make_unique<UniquePicksGenerator>(picks_count, size))
+                       : std::make_unique<NonUniquePicksGenerator>(size);
```

Comment:
ok mabye the function thing was more readable 😆 You can always also just use a declaration and an after

Reply:
O, I'll fix it in `zset_family.cc` too